internal/datastore/spanner/reader.go

```diff
@@ -86,14 +87,28 @@ func queryExecutor(txSource txFactory) common.ExecuteQueryFunc {
 				ResourceAndRelation: &core.ObjectAndRelation{},
 				Subject:             &core.ObjectAndRelation{},
 			}
+			var caveatName spanner.NullString
+			var caveatCtx spanner.NullJSON
 			err := row.Columns(
 				&nextTuple.ResourceAndRelation.Namespace,
 				&nextTuple.ResourceAndRelation.ObjectId,
 				&nextTuple.ResourceAndRelation.Relation,
 				&nextTuple.Subject.Namespace,
 				&nextTuple.Subject.ObjectId,
 				&nextTuple.Subject.Relation,
+				&caveatName,
+				&caveatCtx,
 			)
+			if caveatName.Valid && caveatCtx.Valid {
```

Comment:
Same here, why enforcing caveat context needs to be non-null? I'm surprised no test fails, we may have a gap.

Reply:
Isn't the context nil when it is unspecified?